src/operations/operation.ts

```diff
@@ -75,6 +75,8 @@ export abstract class AbstractOperation<TResult = any> {
     this.trySecondaryWrite = false;
   }
 
+  abstract get commandName(): string;
```

Comment:
q: why are we using a `getter` instead of just a string property?

Reply:
We want the command name to be read-only, and since there's no setter, a getter will ensure this. In addition, across all instances there's only one definition of the getter, but as for a string it would be duplicated for every instance.

Reply:
I guess it's a small optimization, but the getter won't be called on every instance necessarily (for ex: if logging is turned off), so it's not necessary to duplicate was my logic. @nbbeeken anything to add?

Reply:
@nbbeeken just sent me some context, here is a prior precedent for constant string names as getters in our codebase: https://github.com/mongodb/node-mongodb-native/blob/NODE-5741-use-io-async/src/error.ts#L688-L690

Reply:
Not much, getter vs property is borderline stylistic, I think getters seem natural for this kind of const value and it does seem to follow our error classes and _bsontype approaches.